### PR TITLE
Add test that validates fingerprint.StartsWith when read from files

### PR DESCRIPTION
**Pull request by djaglowski**

### CODE REVIEW

File: operator/builtin/input/file/fingerprint_test.go [hunk 1]
```diff
@@ -212,4 +214,49 @@ func TestFingerprintStartsWith(t *testing.T) {
 	}
 }
 
+func TestFingerprintStartsWith_FromFile(t *testing.T) {
+	rand.Seed(112358)
```
Comment (tigrannajaryan) [inline operator/builtin/input/file/fingerprint_test.go:218]: This global may not work well if tests are run in parallel (do we?).
`r := rand.New(rand.NewSource(112358))` and using `r.Read()` should be safer.

File: operator/builtin/input/file/fingerprint_test.go [hunk 2]
```diff
@@ -212,4 +214,49 @@ func TestFingerprintStartsWith(t *testing.T) {
 	}
 }
 
+func TestFingerprintStartsWith_FromFile(t *testing.T) {
```
Comment (tigrannajaryan) [inline operator/builtin/input/file/fingerprint_test.go:217]: I am not familiar with what the fingerprints and how they work so I don't understand what this test does. Perhaps that's fine and I am not supposed to understand. If you want me to understand though please add some comments to explain what it does :-)

**Review by tigrannajaryan — APPROVED** [review]

File: operator/builtin/input/file/fingerprint_test.go (hunk 1 above)
Comment (djaglowski) [reply operator/builtin/input/file/fingerprint_test.go]: Good catch. Fixed.

File: operator/builtin/input/file/fingerprint_test.go (hunk 2 above)
Comment (djaglowski) [reply operator/builtin/input/file/fingerprint_test.go:224]: It's certainly not meant to be a secret. :) 

I expanded the comments on the `Fingerprint` struct and its `StartsWith` method, and also added a high level explanation on the new test.

**Review by tigrannajaryan — APPROVED** [review]
Thank you for the comments, very helpful.

**Pull request merged**